### PR TITLE
change the managed serviceaccount to a template type addon

### PR DESCRIPTION
**Pull request by zhujian7**
issue reference: https://issues.redhat.com/browse/ACM-7067

This PR will move the managed service account to a [template-type](https://open-cluster-management.io/developer-guides/addon/#build-an-addon-with-addon-template) addon, so there will be no managed service account addon manager running on the hub cluster, instead, we add an addonTemplate CR on the hub. 
- remove managed service account addon manager deployment and add addon addonTemplate CR on the hub
- change the generate charts script to generate charts for template-type addons
- update permission for addontemplate and addondeploymentconfig

### CODE REVIEW

**Comment by zhujian7** [comment]
@qiujian16 @xuezhaojun could you help review this?

File: pkg/templates/charts/toggle/managed-serviceaccount/templates/addontemplate.yaml [hunk 1]
```diff
@@ -0,0 +1,135 @@
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: AddOnTemplate
+metadata:
+  name: managed-serviceaccount
```
Comment (qiujian16) [inline pkg/templates/charts/toggle/managed-serviceaccount/templates/addontemplate.yaml:4]: should you have the version in the name?

File: hack/bundle-automation/charts-config.yaml [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+  #     chart-path: "charts/managed-serviceaccount"
+  #     always-or-toggle: "toggle"
+  #     imageMappings:
+  #       managed-serviceaccount: managed_serviceaccount
+  #     inclusions:
+  #       - "pullSecretOverride"
+  template_addons:
     - name: "managed-serviceaccount"
-      chart-path: "charts/managed-serviceaccount"
+      yaml-paths:
```
Comment (qiujian16) [inline hack/bundle-automation/charts-config.yaml:15]: actually why we use yaml but not helm chart?

File: pkg/templates/charts/toggle/managed-serviceaccount/templates/addontemplate.yaml (hunk 1 above)
Comment (zhujian7) [reply pkg/templates/charts/toggle/managed-serviceaccount/templates/addontemplate.yaml:4]: `managed-serviceaccount-2.4` ?

File: hack/bundle-automation/charts-config.yaml (hunk 2 above)
Comment (zhujian7) [reply hack/bundle-automation/charts-config.yaml:15]: the manifest files we used to deploy the agent upstream are yaml-format, which does not support helm format.

File: hack/unit-test-crds/addontemplates.yaml [hunk 3]
```diff
@@ -0,0 +1,472 @@
+# copied from https://github.com/open-cluster-management-io/api/blob/main/addon/v1alpha1/0000_03_addon.open-cluster-management.io_addontemplates.crd.yaml
```
Comment (xuezhaojun) [inline hack/unit-test-crds/addontemplates.yaml:1]: Where is this yaml used? I can't find by global search the key "unit-test-crds/addontemplates.yaml"

**Comment by xuezhaojun** [comment]
Is addonTemplate should join this rbac file? https://github.com/stolostron/backplane-operator/blob/7b9cbb638fc2d36fa0a5ce5a5f4cb2c092fd330a/pkg/templates/charts/always/rbac-aggregates/templates/clusterrole-admin-clc.yaml#L44

**Review by xuezhaojun — COMMENTED** [review]
Great work!  The changes look good to me.

**Comment by zhujian7** [comment]
Thanks @xuezhaojun, Do you know if this file is generated by scripts or manually?

**Comment by xuezhaojun** [comment]
> Thanks @xuezhaojun, Do you know if this file is generated by scripts or manually?

I viewed the history of this file it seems manually added, in [klusterletconfig PR](https://github.com/stolostron/backplane-operator/pull/491) I added it manually.

This [placement PR](https://github.com/stolostron/backplane-operator/pull/132) from @haoqing0110  also seems manually changed.

**Comment by zhujian7** [comment]
@xuezhaojun @qiujian16 PTAL

**Comment by zhujian7** [comment]
Hi @dislbenn, could you help review this?

**Comment by xuezhaojun** [comment]
> @xuezhaojun @qiujian16 PTAL

Nothing update from me, good work!

**Comment by dislbenn** [comment]
/lgtm
/hold

@zhujian7 will these changes impact this story: https://issues.redhat.com/browse/ACM-7008? If not, we can unhold this PR. If yes, can you add a comment to that issue?

**Comment by zhujian7** [comment]
> will these changes impact this story: https://issues.redhat.com/browse/ACM-7008?
@dislbenn, thanks for the review, As far as I understand, there should be no impact. I just refactored how the managed service account addon is installed(deployment -> addontemplate), its functionality has not changed.

**Pull request merged**